### DISCUSSION TITLE
Is it possible to call dist.all_reduce manually in train_step?

### DISCUSSION THREAD

**Discussion by sandylaker** [Lightning-AI/pytorch-lightning#7693] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule] [answered]
In my code, I would like to synchronize a tensor across all the gpus in `train_step`, which is a temporary variable. Is it allowed to call `torch.distributed.all_reduce` in this case? Or there is a specific function in `pytorch_lightning` that does the job?

**Comment by SeanNaren** [comment, marked as answer]
Hey @sandylaker! You can use `torch.distributed.all_reduce`. There is also within the `LightningModule` this function, however it may be better to expose this within lightning to make it easier to access:

```python
x = self.trainer.accelerator.training_type_plugin.reduce(x)
```
